### PR TITLE
Update Azure.ResourceManager.RecoveryServicesSiteRecovery SDK to 2022-10-01

### PR DESCRIPTION
**Pull request by shesamian**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

This PR was generated from [stable 2022-10-01](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2022-10-01/service.json) of recoveryservicessiterecovery

[`Microsoft.Azure.Management.RecoveryServices.SiteRecovery`](https://www.nuget.org/packages/Microsoft.Azure.Management.RecoveryServices.SiteRecovery/) has been deprecated according to the readme so I didn't update it. I only updated [Azure.ResourceManager.RecoveryServicesSiteRecovery](https://www.nuget.org/packages/Azure.ResourceManager.RecoveryServicesSiteRecovery/).

### CODE REVIEW

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/CHANGELOG.md [hunk 1]
```diff
@@ -1,15 +1,16 @@
 # Release History
 
-## 1.0.0-beta.2 (Unreleased)
+## 1.0.0-beta.2 (2022-10-01)
```
Comment (ArthurMa1978) [inline sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/CHANGELOG.md:3]: The date here is for which day you expected to release the new version, not the api version

File: sdk/recoveryservices-siterecovery/Microsoft.Azure.Management.RecoveryServices.SiteRecovery/src/Microsoft.Azure.Management.RecoveryServices.SiteRecovery.csproj [hunk 2]
```diff
@@ -6,7 +6,7 @@
   <PropertyGroup>
     <PackageId>Microsoft.Azure.Management.RecoveryServices.SiteRecovery</PackageId>
     <Description>Provides developers with libraries for the updated recovery services site recovery feature under Azure Resource manager.</Description>
-    <Version>2.2.0-preview</Version>
+    <Version>2.2.1-preview</Version>
```
Comment (ArthurMa1978) [inline sdk/recoveryservices-siterecovery/Microsoft.Azure.Management.RecoveryServices.SiteRecovery/src/Microsoft.Azure.Management.RecoveryServices.SiteRecovery.csproj:9]: Track 1 is being deprecated, so dont need update this old sdk except you want to use this code for PowerShell cmdlets

Comment (shesamian) [reply sdk/recoveryservices-siterecovery/Microsoft.Azure.Management.RecoveryServices.SiteRecovery/src/Microsoft.Azure.Management.RecoveryServices.SiteRecovery.csproj]: Reverted changes to "Track 1"

File: sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/CHANGELOG.md (hunk 1 above)
Comment (shesamian) [reply sdk/recoveryservices-siterecovery/Azure.ResourceManager.RecoveryServicesSiteRecovery/CHANGELOG.md]: I set it to `2023-01-10`

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**